pyaedt/modeler/cad/component_array.py

```diff
@@ -0,0 +1,843 @@
+from __future__ import absolute_import
+
+from collections import OrderedDict
+import os
+import re
+
+from pyaedt import pyaedt_function_handler
+from pyaedt.generic.general_methods import _uname
+from pyaedt.generic.general_methods import read_csv
+
+
+class ComponentArray(object):
+    """Manages object attributes for 3DComponent array.
+
+    Parameters
+    ----------
+    app : :class:`pyaedt.Hfss`
+        Hfss pyaedt object.
+    name : str, optional
+        Array name. The default value is ``None``.
+    props : dict, optional
+        Dictionary of properties. The default value is ``None``.
+
+    Examples
+    --------
+    Basic usage demonstrated with an HFSS design with an existing array:
+
+    >>> from pyaedt import Hfss
+    >>> aedtapp = Hfss(projectname="Array.aedt")
+    >>> array_names = aedtapp.component_array_names[0]
+    >>> array = aedtapp.component_array[array_names[0]]
+    """
+
+    def __init__(self, app, name=None, props=None):
+        if name:
+            self._m_name = name
+        else:
+            self._m_name = _uname("Array_")
+
+        self._app = app
+
+        self._logger = app.logger
+
+        self._omodel = self._app.get_oo_object(self._app.odesign, "Model")
+
+        self._oarray = self._app.get_oo_object(self._omodel, name)
+
+        # Data that can not be obtained from CSV
+        try:
+            self._cs_id = props["ArrayDefinition"]["ArrayObject"]["ReferenceCSID"]
+        except AttributeError:  # pragma: no cover
+            self._cs_id = 1
+
+        self._array_info_path = None
+
+        self._update_cells = True
+
+        if self._app.settings.aedt_version > "2023.2":  # pragma: no cover
+            self.export_array_info(array_path=None)
+            self._array_info_path = os.path.join(self._app.toolkit_directory, "array_info.csv")
+
+        self._cells = None
+
+        self._post_processing_cells = {}
+
+    @property
+    def component_names(self):
+        """List of component names.
+
+        Returns
```

Comment:
I don't think properties need to have "Returns" sections. The property description should provide all the information needed. Please remove the "Returns" sections for the properties.